platform/windows_x64/docker/install_msys2.ps1

```diff
@@ -0,0 +1,85 @@
+# changes every couple days. We're about to install the latest version of
+# a bunch of packages with pacman anyway, so we've already lost. pacman provides
+# little control for installing specific package versions.
+Write-Verbose "Download msys2"
+Invoke-WebRequest `
+  -Uri 'https://github.com/msys2/msys2-installer/releases/download/nightly-x86_64/msys2-base-x86_64-latest.sfx.exe' `
+  -OutFile "${env:EF_INSTALL_TEMP_DIR}/msys2-base-x86_64-latest.sfx.exe"
+
+# Run the installer. It's a self-extracting archive. It extracts to .\msys64.
+#   -y     skip prompts
+#   -oC:\  write to C:\ instead of working directory
+Write-Verbose "Extract msys2"
+& "${env:EF_INSTALL_TEMP_DIR}/msys2-base-x86_64-latest.sfx.exe" -y -oC:\
+
+# Add msys2 to PATH.
+$env:PATH = [Environment]::GetEnvironmentVariable('PATH', 'Machine')
+$env:PATH = "$env:PATH;C:\msys64\usr\bin"
+[Environment]::SetEnvironmentVariable('PATH', $env:PATH, 'Machine')
+
+# Install C/C++ toolchains.
+# See https://www.msys2.org/docs/environments/.
+#
+# - gcc: Bazel auto-configures this as msys-gcc.
+#   Installed in C:\msys64\usr\bin. Uses the cygwin C runtime.
+#   Useful for GNU compatibility and simple configuration.
+# - mingw-w64-x86_64-gcc: Bazel auto-configures this as mingw-gcc.
+#   Installed in C:\msys64\mingw64\usr\bin. Uses the MSVCRT runtime.
+#   rules_foreign_cc needs this in some cases for GNU compatibility.
+# - mingw-w64-ucrt-x86_64-gcc: A GCC toolchain.
+#   Installed in C:\msys64\ucrt64\usr\bin, uses UCRT64.
+#   Should be compatible with libraries built with MSVC linked against UCRT64.
+#   We need to configure our own toolchains for this.
+# - mingw-w64-ucrt-x86_64-clang: A clang toolchain.
+#   Installed in C:\msys64\ucrt64\usr\bin, uses UCRT64.
+#   Should be compatible with libraries built with MSVC linked against UCRT64.
+#   We need to configure our own toolchains for this.
+#
+# NOTE: we invoke pacman via bash. It seems to depend on the environment
+# to initialize gnupg correctly and to not trigger the Docker bug
+# https://github.com/microsoft/hcsshim/issues/696.
+#
+# bash -l: run a login shell, evaluting startup files.
+# bash -c: run a command.
+#
+# pacman --sync: install packages, analogous to apt-get.
+# pacman --refresh: download latest package metadata.
+# pacman --sysupgrade: upgrade installed packages.
+# pacman --needed: do not reinstall up-to-date packages.
+Write-Verbose "Install msys2 C/C++ toolchains"
+bash -l -c "pacman --sync --refresh --sysupgrade --needed --noconfirm --noprogressbar"
+bash -l -c "pacman --sync --needed --noconfirm --noprogressbar gcc mingw-w64-x86_64-gcc mingw-w64-ucrt-x86_64-gcc mingw-w64-ucrt-x86_64-clang"
+
+# Clean up temporary files created by pacman.
+# This may be needed to avoid triggering the Docker bug 
+# https://github.com/microsoft/hcsshim/issues/696.
+# We don't want these files in the image anyway.
+bash -l -c "rm -rf /C/Users/ContainerUser/* /var/cache/pacman/pkg/*"
```

Comment:
```suggestion
bash -l -c "rm -rf $(cygpath C:\\Users\\ContainerUser)* /var/cache/pacman/pkg/*"
```

Reply:
I think this change will probably work, but I'm afraid to touch it. This part was extremely delicate, and right now I can't afford the time to manually verify a change here.